sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -345,8 +352,30 @@ public void finish(@Nullable SpanStatus status) {
   @Override
   @ApiStatus.Internal
   public void finish(@Nullable SpanStatus status, @Nullable SentryDate finishDate) {
+
+    // try to get the high precision timestamp from the root span
+    SentryDate finishTimestamp = root.getFinishDate();
+
+    // if a finishDate was passed in, use that instead
+    if (finishDate != null) {
+      finishTimestamp = finishDate;
+    }
+
+    // if it's not set -> fallback to the current time
+    if (finishTimestamp == null) {
+      finishTimestamp = hub.getOptions().getDateProvider().now();
+    }
+
+    // auto-finish any idle spans first
+    for (final Span span : children) {
+      if (span.getOptions().isIdle()) {
+        span.finish(getSpanContext().status, finishTimestamp);
```

Comment:
hrm, is there some idea behind not using the `status` parameter, which we are finishing this transaction with?